src/care/config.h.in

```diff
@@ -30,21 +32,58 @@
 #define CARE_DLL_API
 #endif // defined(_WIN32) && !defined(CARESTATICLIB)
 
+#if defined(ENABLE_CUDA)
```

Comment:
I'm wondering if we should move a bunch of this to a different header file called gpu_config.h or GPUMacros.h or something. What do you think?

Reply:
Also, I'm hesitant about including the cuda and hip header files since that means they will literally get included in almost every single source file. We were burned by this before with curand.

Reply:
If \_\_HIPCC\_\_ is not defined by the compiler, then we should ask Kostas to get that changed so we don't have to include these headers.

Reply:
I emailed Kostas regarding __HIPCC__ . The issue appears to be well known, see https://github.com/ROCm-Developer-Tools/HIP/issues/29 . I ran into it when working on my CHAI branch . I sent an email to Kostas. In any case, I'll look at re-factoring the headers a bit, see what I can do

Reply:
Nevermind, I was confused. 

In CARE, this is not an issue, since we compile with hipcc and all these macros etc are defined by the compiler. In CHAI, which compiles with a regular c++ compiler and then includes hip_runtime, things are different. In that case we need to #include <hip/hip_runtime.h> (or hip_runtime_api or hip_common, which may be lighter weight). 

But since this is a CARE pull request and CARE compiles with hipcc, we don't need to worry about including the header

Reply:
Added a new file for GPU macros. It looks a bit cleaner, better organized.

For now, I just add #include "care/GPUMacros.h" to the bottom of config.h . Adding it to every file is a pain, and things would behave badly if we forgot to include it in a new file that someone made. I think it is better to just put it within config.h since that is already included everywhere.